grin/test/Transformations/ExtendedSyntax/MangleNamesSpec.hs

```diff
@@ -32,22 +32,39 @@ spec = do
       |]
     (mangleNames before) `sameAs` after
 
+  it "as-pattern" $ do
+    let before = [prog|
+      grinMain =
+        n0 <- pure 5
+        n1@n2 <- pure n0
+        v@(CInt k) <- pure (CInt n0)
+        pure k
+      |]
+    let after = [prog|
+      name.0 =
+        name.1 <- pure 5
+        name.2@name.3 <- pure name.1
```

Comment:
I think this is very confusing. I agree that the Node Binding should have a unique name via the as pattern. But I see only complication having an as pattern for a variable.

Reply:
We will restrict the available patterns only to nodes in the future. I added a few lines about this to #32.